Tests/BSONTests/ExtendedJSONConversionTests.swift

```diff
@@ -4,7 +4,75 @@ import Nimble
 import NIO
 import XCTest
 
+extension CodingUserInfoKey {
+    static let barInfo = CodingUserInfoKey(rawValue: "bar")!
+}
+
 open class ExtendedJSONConversionTestCase: BSONTestCase {
+    struct Test: Codable, Equatable {
+        let x: Bool
+        let y: Int32
+        let z: BSONRegularExpression
+    }
+
+    struct Foo: Codable, Equatable {
```

Comment:
can we move this struct definition into the test that uses it? It will help organize this file a little bit. Ditto for the `Test` struct.

Reply:
done